proposals/collection-expression-arguments.md

```diff
@@ -268,3 +268,147 @@ strongly triggers the view that this is simply an implicit-object-creation.  And
 case where a constructor *is* actually called (like for `Dictionary<,>`) it is misleading when calling a *create
 method*, or creating an interface.  Finally, there is general apprehension around using `new` at all as there
 is a feeling of redundancy around both the `new` indicating a new instance, *and* `[...]` indicating a new instance.
+
+## Conversions
+
+Collection arguments are *not* considered when determining *collection expression* conversions.
+
+## Construction
+
+Construction is updated as follows.
+
+The elements of a collection expression are evaluated in order, left to right.
+Within *collection arguments*, the arguments are evaluated in order, left to right.
+Each element or argument is evaluated exactly once, and any further references refer to the results of this initial evaluation.
+
+If *collection_arguments* is included and is not the first element in the collection expression, a compile-time error is reported.
+
+If the target type is a *struct* or *class type* that implements `System.Collections.IEnumerable`, and the target type does not have a *create method*, and the target type is not a *generic parameter type* then:
+* [*Overload resolution*](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1264-overload-resolution) is used to determine the best instance constructor from the *argument list*.
+  * If the *argument list* contains any values with *dynamic* type, the best instance constructor is determined at runtime.
+* If a best instance constructor is found, the constructor is invoked with the *argument list*.
+  * If the constructor has a `params` parameter, the invocation may be in expanded form.
+* Otherwise, a binding error is reported.
+
+```csharp
+// List<T> candidates:
+//   List<T>()
+//   List<T>(IEnumerable<T> collection)
+//   List<T>(int capacity)
+List<int> l;
+l = [with(capacity: 3), 1, 2]; // new List<int>(capacity: 4)
+l = [with([1, 2]), 3];         // new List<int>(IEnumerable<int> collection)
+l = [with(default)];           // error: ambiguous constructor
```

Comment:
ambiguous in the case where there are multiple candidates.  but would be ok if there wasn't, right?